### ISSUE TITLE
Stopping Runtimes Crashes Obsidian

### ISSUE THREAD

**Issue by mayurankv** [twibiral/obsidian-execute-code#188]
When I stop a runtime in the runtimes side pane, my Obsidian closes unexpectedly. No clue why this is. I have a reasonable number of plugins so could be a conflict somewhere. Would appreciate any help debugging. Mac OS Ventura - Obsidian v1.0.3 - Execute Code v1.5.0

**Comment by chlohal** [comment]
This is troubling. What languages are you stopping? 

If you're having this problem with all Notebook Mode languages (Python, Node, & R), please try entering `pkill` in a terminal window and sending a screenshot of the output.

Thanks!

**Comment by mayurankv** [comment]
Python is the current offender, let me try a javascript block.

**Comment by mayurankv** [comment]
Huh, this is really weird now. I can't even see python or shell runtimes but I can see javascript. The codeblocks work but I see no runtime. I actually am not familiar enough with any other language to type up any code. I will try and find some snippets to test and report back.

Also killing javascript does cause this problem too.

When I kill a shell in activity monitor or with pkill, I just get an Error! popup. Not sure how to identify the PID of a python script?

**Comment by mayurankv** [comment]
No clue what changed, but I can now see python runtimes again. Stopping them still crashes Obsidian. Still can't see shell runtimes though (though this may be expected if shells don't need a runtime?).

**Comment by chlohal** [comment]
Yep, only Javascript, Python, and R have runtimes. All other blocks run once & then close their runtime.

**Comment by chlohal** [comment]
> Huh, this is really weird now. I can't even see python or shell runtimes but I can see javascript.

Do you have Notebook Mode on for Python? The setting should look like this (circled):

![image](https://user-images.githubusercontent.com/26300854/205761447-56c64380-1eb3-4103-a496-5ae4e5f124dc.png)

**Comment by mayurankv** [comment]
I did flick it on and off which must have been why the python ones disappeared. Still crashing on kill though unfortunately.

**Comment by mayurankv** [comment]
Returned to this issue and killing the runtimes no longer works at all (though nothing crashes). I'm currently getting the error:

```
node:internal/errors:841 Uncaught (in promise) Error: Command failed: pkill -P 85436
    at __node_internal_genericNodeError (node:internal/errors:841:15)
    at checkExecSyncError (node:child_process:837:11)
    at execSync (node:child_process:908:15)
    at node:electron/js2c/asar_bundle:5:12704
    at killWithChildren_default (plugin:execute-code:12174:39)
    at eval (plugin:execute-code:12242:7)
    at new Promise (<anonymous>)
    at PythonExecutor.stop (plugin:execute-code:12238:12)
    at HTMLButtonElement.eval (plugin:execute-code:12770:53)
```

Killing either javascript or python runtimes gives this. No clue why unfortunately.

**Comment by mayurankv** [comment]
As does the R runtime have the same error :((.

**Comment by mayurankv** [comment]
I believe this could be resolved by letting the user change the pkill command in settings. As in to /usr/bin/pkill. Using just the python command for python doesn't work for me, I have to use the result of which python so similarly I think the pkill is causing issues.

**Comment by twibiral** [comment]
Fixed with new release [version 1.12.0](https://github.com/twibiral/obsidian-execute-code/releases/tag/1.12.0)
